feat(ux): `across`/`if_any`/`if_all` selectors for whole-table operations (#5598)
ibis-project/ibis: docs/blog/selectors.md

│ Adelie  │ Torgersen │            nan │           nan │                 ∅ │           ∅ │ ∅      │  2007 │
│ Adelie  │ Torgersen │           36.7 │          19.3 │               193 │        3450 │ female │  2007 │
│ Adelie  │ Torgersen │           39.3 │          20.6 │               190 │        3650 │ male   │  2007 │
│ Adelie  │ Torgersen │           38.9 │          17.8 │               181 │        3625 │ female │  2007 │
│ Adelie  │ Torgersen │           39.2 │          19.6 │               195 │        4675 │ male   │  2007 │
│ Adelie  │ Torgersen │           34.1 │          18.1 │               193 │        3475 │ ∅      │  2007 │
│ Adelie  │ Torgersen │           42.0 │          20.2 │               190 │        4250 │ ∅      │  2007 │
│ …       │ …         │              … │             … │                 … │           … │ …      │     … │
└─────────┴───────────┴────────────────┴───────────────┴───────────────────┴─────────────┴────────┴───────┘
```

## Examples

### Normalization

Let's say you want to compute the z-score of every numeric column and replace
the existing data with that normalized value. Here's how you'd do that with

> selectors:

Reviewer: This suggestion, though, might be more appropriate for a tutorial and less for a blog post, feel free to ignore!

> selectors:

Author: Is a link to wikipedia a solid middle ground?

│ Gentoo  │ Biscoe │      -0.975022 │     -0.287513 │         -0.491442 │   -0.448342 │ female │  2009 │
│ Gentoo  │ Biscoe │       0.387793 │     -0.898997 │         -1.108253 │   -1.241809 │ female │  2007 │
│ Gentoo  │ Biscoe │       0.809616 │      0.222056 │          0.125368 │    1.237778 │ male   │  2007 │
│ Gentoo  │ Biscoe │       0.030865 │     -0.491341 │         -0.337240 │    0.642677 │ male   │  2007 │
│ Gentoo  │ Biscoe │      -0.326062 │     -1.510481 │         -1.108253 │   -1.043442 │ female │  2007 │
│ Gentoo  │ Biscoe │      -0.682990 │     -0.389427 │         -0.954050 │   -0.547525 │ female │  2007 │
│ Gentoo  │ Biscoe │      -0.261167 │      0.323970 │          0.279571 │    0.245943 │ male   │  2007 │
│ Gentoo  │ Biscoe │      -1.364397 │     -1.612395 │         -1.262455 │   -1.340993 │ female │  2007 │
│ Gentoo  │ Biscoe │      -0.228719 │      0.425884 │         -0.337240 │    0.146759 │ male   │  2007 │
│ …       │ …      │              … │             … │                 … │           … │ …      │     … │
└─────────┴────────┴────────────────┴───────────────┴───────────────────┴─────────────┴────────┴───────┘
```

> Since this "desugars" into a plain selection as if you had called `select` or

Reviewer: ```suggestion
Since ibis translates this into a plain selection as if you had called `select` or
```

│ adelie  │ torgersen │            nan │           nan │                 ∅ │           ∅ │ ∅      │  2007 │
│ adelie  │ torgersen │           36.7 │          19.3 │               193 │        3450 │ female │  2007 │
│ adelie  │ torgersen │           39.3 │          20.6 │               190 │        3650 │ male   │  2007 │
│ adelie  │ torgersen │           38.9 │          17.8 │               181 │        3625 │ female │  2007 │
│ adelie  │ torgersen │           39.2 │          19.6 │               195 │        4675 │ male   │  2007 │
│ adelie  │ torgersen │           34.1 │          18.1 │               193 │        3475 │ ∅      │  2007 │
│ adelie  │ torgersen │           42.0 │          20.2 │               190 │        4250 │ ∅      │  2007 │
│ …       │ …         │              … │             … │                 … │           … │ …      │     … │
└─────────┴───────────┴────────────────┴───────────────┴───────────────────┴─────────────┴────────┴───────┘
```

### Multiple Computations per Column

What if I want to compute multiple things? Heck yeah!

```
In [26]: t.group_by("sex").mutate(

>     ...:     s.across(s.numeric() & ~s.c("year"), dict(cent=_ - _.mean(), zscore=(_ - _.mean()) / _.std()))

Reviewer: ```suggestion
    ...:     s.across(s.numeric() & ~s.c("year"), dict(center=_ - _.mean(), zscore=(_ - _.mean()) / _.std()))
```

Reviewer: Hmm, I guess below you call this `centered` -- either is good, but nice to be consistent.

Author: I did this this cut down the width of the table a bit. I'll change the other to `cent`.

Author: Went with `centered` and just used another selector to subset some columns :)

│ Adelie  │ Dream  │           40.3 │          18.5 │               196 │        4350 │ male   │  2008 │                0.187500 │              -0.387500 │                        1.0 │ … │
│ Adelie  │ Dream  │           43.2 │          18.5 │               192 │        4100 │ male   │  2008 │                3.087500 │              -0.387500 │                       -3.0 │ … │
│ Adelie  │ Dream  │           39.5 │          16.7 │               178 │        3250 │ female │  2007 │                1.644444 │              -1.144444 │                       -7.0 │ … │
│ Adelie  │ Dream  │           39.5 │          17.8 │               188 │        3300 │ female │  2007 │                1.644444 │              -0.044444 │                        3.0 │ … │
│ …       │ …      │              … │             … │                 … │           … │ …      │     … │                       … │                      … │                          … │ … │
└─────────┴────────┴────────────────┴───────────────┴───────────────────┴─────────────┴────────┴───────┴─────────────────────────┴────────────────────────┴────────────────────────────┴───┘
```

### Filtering Selectors

> You can also express complex filters more concise.

Reviewer: ```suggestion
You can also express complex filters more concisely.
```

> Let's say we only want to keep rows where all the bill size related columns' absolute values are greater than 2.

Reviewer: bill size z_score related columns?

Author: SGTM!
